src/Draco.Compiler.Tests/Syntax/ParserTests.cs

```diff
@@ -114,4 +204,1269 @@ public void TestEmptyFuncWithoutClosingCurly()
+            this.T(TokenType.MultiLineStringEnd);
+        }
+        var quotes = "\"\"\"";
+        this.StringTestsPlaceHolder($$"""
+            {{quotes}}
+            Hello, 
+            \{"World!"
+            {{quotes}}
+            """, UnclosedString);
+    }
+
+    [Fact]
+    public void TestDeclarationMissingSemicolon()
+    {
+        void Declaration()
+        {
+            this.N<Stmt.Decl>();
+            {
+                this.N<Decl.Variable>();
+                {
+                    this.T(TokenType.KeywordVal);
+                    this.T(TokenType.Identifier);
+                    this.N<ValueInitializer>();
+                    {
+                        this.T(TokenType.Assign);
+                        this.N<Expr.Literal>();
+                        {
+                            this.T(TokenType.LiteralInteger);
+                        }
+                    }
+                    this.MissingT(TokenType.Semicolon);
+                }
+            }
+        }
+        this.MainFunctionPlaceHolder("val x = 5", Declaration);
+    }
+
+    [Fact]
+    public void TestDeclarationNoTypeNoValue()
+    {
+        void Declaration()
+        {
+            this.N<Stmt.Decl>();
+            {
+                this.N<Decl.Variable>();
+                {
+                    this.T(TokenType.KeywordVal);
+                    this.T(TokenType.Identifier);
+                    this.T(TokenType.Semicolon);
+                }
+            }
+        }
+        this.MainFunctionPlaceHolder("val x;", Declaration);
+    }
+
+    [Fact]
+    public void TestDeclarationNoTypeValue()
+    {
+        void Declaration()
+        {
+            this.N<Stmt.Decl>();
+            {
+                this.N<Decl.Variable>();
+                {
+                    this.T(TokenType.KeywordVal);
+                    this.T(TokenType.Identifier);
+                    this.N<ValueInitializer>();
+                    {
+                        this.T(TokenType.Assign);
+                        this.N<Expr.Literal>();
+                        {
+                            this.T(TokenType.LiteralInteger);
+                        }
+                    }
+                    this.T(TokenType.Semicolon);
+                }
+            }
+        }
+        this.MainFunctionPlaceHolder("val x = 5;", Declaration);
+    }
+
+    [Fact]
+    public void TestDeclarationNoTypeValueMissingValue()
+    {
+        void Declaration()
+        {
+            this.N<Stmt.Decl>();
+            {
+                this.N<Decl.Variable>();
+                {
+                    this.T(TokenType.KeywordVal);
+                    this.T(TokenType.Identifier);
+                    this.N<ValueInitializer>();
+                    {
+                        this.T(TokenType.Assign);
+                        this.N<Expr.Unexpected>();
+                        {
+                            this.T(TokenType.Semicolon);
```

Comment:
Does the semicolon belong to the unexpected node here? If so, that might be an error.

Reply:
You could have just accidently put it here too.